runtime/src/main/java/com/flipkart/flux/representation/ClientElbPersistenceService.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2012-2018, the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.flux.representation;
+
+import com.flipkart.flux.api.ClientElbDefinition;
+import com.flipkart.flux.clientelb.dao.iface.ClientElbDAO;
+import com.flipkart.flux.domain.ClientElb;
+import com.flipkart.flux.persistence.SelectDataSource;
+import com.flipkart.flux.persistence.Storage;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.transaction.Transactional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <code>ClientElbPersistenceService</code> class does persistence related operations in memory
+ * before storing, updating, reading or deleting objects in DB. It is also responsible for maintaining
+ * an in-memory cache(Google Guava: https://github.com/google/guava/wiki) of clientElb entries for faster query results.
+ *
+ * @author akif.khan
+ */
+@Singleton
+public class ClientElbPersistenceService {
+
+    private ClientElbDAO clientElbDAO;
+
+    private LoadingCache<String, ClientElb> clientElbCache;
+
+    private Integer MAX_CACHE_SIZE;
```

Comment:
Why is this cache limited size ?

Reply:
We are passing the cache size via config file. Though, in our use case it is not required but 
it is a good practice to keep the cache size configurable which makes the testing of cache
operations easier.